multimodal/video_understanding_agent/app.py

```
@@ -0,0 +1,148 @@
+    if action_match:
+        sections["actions"] = action_match.group(1).strip()
+
+    if not any(sections.values()):
+        sections["chapters"] = text.strip()
+
+    return sections
+
+PROMPT = """Analyze this YouTube video and return a structured response with exactly three sections using these headings:
+
+## Chapter-by-Chapter Summary
+List each chapter or major segment with its timestamp (e.g. "0:00 - Introduction", "3:45 - Main concept explained"). Write 2–3 sentences describing what happens in each chapter. If the video has no clear chapters, break it into logical segments with approximate timestamps.
```

Ruff (0.15.10)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)